src/app/api/organizations/[organizationId]/route.ts

```diff
@@ -24,3 +27,34 @@ export async function DELETE(
   }
   return new NextResponse(undefined, { status: 204 });
 }
+
+// @route PUT /api/organizations/[organizationId] - updates an existing organization in the database
+export async function PUT(
+  _request: NextRequest,
+  { params }: { params: { organizationId: string } }
+) {
+  const validationResult = zObjectId.safeParse(params.organizationId);
+  if (!validationResult.success) {
+    return NextResponse.json(
+      { message: 'Invalid Organization Id' },
+      { status: 400 }
+    );
+  }
+
+  // get the updated organization data
+  const requestBody = await _request.json();
```

Comment:
Just as you validate the organization id against `zObjectId`, validate the request body using `zUpdateOrganizationRequest`. When you run `safeParse`, it returns you an object with a `success` prop and a `data` prop. The `data` prop has any extra fields (i.e. `softDelete`) stripped away. Pass in the `data` prop to the server func, and soft delete will not be possible

Reply:
We'll also want a 400 if the body fails validation